File: frontend/src/app/order/order.model.ts
import { Product } from '../products/product/product.model';

export class Order {
    id: number;
    amount: number;
    isOrdered: boolean;
    orderDate: Date;
    userId: number;
    productId: number;
    product: Product;
    sum: number;
    constructor(order?) {
        this.id = order ? order.id : 0;
        this.amount = order ? order.amount : null;
        this.userId = order ? order.userId : null;
        this.productId = order ? order.productId : null;
        this.orderDate = order ? order.orderDate : null;
        this.product = order ? order.product : null;
        this.sum = order ? order.sum : null;
    }
}

File: backend/ViewModels/OrderModel.cs
using System;

namespace TodoApi.ViewModels
{
    public class OrderModel
    {
        public int Id { get; set; }
        public int ProductId { get; set; }
        public ProductModel Product { get; set; }
        public int Amount { get; set; }
        public int Sum { get; set; }
        public bool IsOrdered { get; set; }
        public DateTime OrderDate { get; set; }
        public int userId { get; set; }

    }
}